time-elements.js

```diff
@@ -184,6 +192,66 @@
     }
   };
 
+  RelativeTime.prototype.timeUntil = function() {
+    var ms = this.date.getTime() - (new Date().getTime());
+    return this.timeUntilFromMs(ms);
+  };
+
+  RelativeTime.prototype.timeUntilFromMs = function(ms) {
+    var sec = Math.round(ms / 1000);
+    var min = Math.round(sec / 60);
+    var hr = Math.round(min / 60);
+    var day = Math.round(hr / 24);
+    var month = Math.round(day / 30);
+    var year = Math.round(month / 12);
+    if (month >= 18) {
+      return year + ' years from now';
+    } else if (month >= 12) {
+      return 'a year from now';
+    } else if (day >= 45) {
+      return month + ' months from now';
+    } else if (day >= 30) {
+      return 'a month from now';
+    } else if (hr >= 36) {
+      return day + ' days from now';
+    } else if (hr >= 24) {
+      return 'a day from now';
+    } else if (min >= 90) {
+      return hr + ' hours from now';
+    } else if (min >= 45) {
+      return 'an hour from now';
+    } else if (sec >= 90) {
+      return min + ' minutes from now';
+    } else if (sec >= 45) {
+      return 'a minute from now';
+    } else if (sec >= 10) {
+      return sec + ' seconds from now';
```

Comment:
This range is sensitive to a user's browser clock skew not matching the server clock that generated the element. For example, when you submit a comment via a form, and the comment partial is returned from the server, you may see that you wrote your comment "11 seconds from now".

I think we should treat anything under the 90 second range as the "just now" label to give us some buffer between clock differences.